vllm/engine/arg_utils.py

```diff
@@ -796,8 +797,15 @@ def add_cli_args(parser: FlexibleArgumentParser) -> FlexibleArgumentParser:
             "--enable-reasoning",
             action="store_true",
             default=False,
-            help="Whether to enable reasoning_content for the model. "
-            "If enabled, the model will be able to generate reasoning content."
+            help=
+            "[DEPRECATED] " \
+            "The --enable-reasoning flag is deprecated as of v0.8.6. "
+            "Use --reasoning-parser to specify " \
+            "the reasoning parser backend instead. "
+            "This flag (--enable-reasoning) will be " \
+            "removed in a future version. "
```

Comment:
The general rule we are using is 2 SemVer minor releases before removal. So since we deprecate it in `v0.8.x` we can remove it in `v0.10.0`

```suggestion
            "removed in v0.10.0. "
```

Reply:
Done.